bundles/io/org.eclipse.smarthome.io.net/src/main/java/org/eclipse/smarthome/io/net/http/internal/WebClientFactoryImpl.java

```diff
@@ -122,11 +120,16 @@ protected void deactivate() {
     }
 
     @Override
-    public HttpClient createHttpClient(String consumerName, String endpoint) {
-        Objects.requireNonNull(endpoint, "endpoint must not be null");
-        logger.debug("http client for endpoint {} requested", endpoint);
+    @Deprecated
+    public HttpClient createHttpClient(@NonNull String consumerName, @NonNull String endpoint) {
+        return createHttpClient(consumerName);
```

Comment:
Err, by deprecating the feature, the idea was to keep it working just as before - i.e. keep using the (also deprecated) `TrustManagerProvider`. Otherwise, this would break the backward compatibility and if we break it, there wouldn't be much sense to keep the method as deprecated in place.